game/transfers.py

```diff
@@ -166,9 +166,9 @@ def find_escape_route(self) -> Optional[ControlPoint]:
             return self.transport.find_escape_route()
         return None
 
-    def proceed(self) -> None:
-        if not self.destination.is_friendly(self.player):
-            logging.info(f"Transfer destination {self.destination} was captured.")
+    def possible(self) -> bool:
+        if self.transport and not self.next_stop.is_friendly(self.player):
```

Comment:
Disbanding should happen even if `self.transport is None`, otherwise airlifts that were waiting for aircraft will never be pruned.

Reply:
thanks i was already wondering why transport could be none! i added the check for transport not none and destination is friendly as an airlift only has a destination

Reply:
I think `next_stop` being `None` shouldn't cause the _transfer_ to cancel, since there might be an alternative route. With this change the whole transfer will cancel if the _best_ route is cutoff.

This is aiming to fix the case where the `next_stop` was captured, but the destination was still available, right? I think for that case `process` should check that the next stop is invalid, make no progress that turn, and allow the replan. Probably will need to add a check in `PendingTransfers.arrange_transport` to cancel the transfer when there is _no_ path between the current location and the destination, but that should be rare.

Reply:
@DanAlbert  i changed the logic behind the checks a bit to first check if the transit network has a valid link between `position` and `destination` once the `next_stop` gets captured. 
The current logic of the game always calculates the optimum route during `arrange_transport` and creates a transport for the route `transfer.position` to `transfer.next_stop` or `transfer.destination` depending on the possible `link_type`. 
So i only check for `next_stop` in the validation and set `transport` to `None` whenever the `next_stop` gets captured and a link in the transit network still exists. As the `arrange_transport` will be triggered afterwards it will again calculate an optimum route and will setup a new transport. In my testing it mostly just setups an airlift instead because no road network is available anymore. But i think its really rare that there will be an alternative road network as this needs a really complicated conflict theater with lots of CPs. But in total the logic of the code can handle it :)

I think i covered any possibility with this. May i have overseen sth?

Reply:
> In my testing it mostly just setups an airlift instead because no road network is available anymore.

Yeah, that should be typical for most campaigns.

> But i think its really rare that there will be an alternative road network as this needs a really complicated conflict theater with lots of CPs.

These complicated cases were a lot more common before. Those scenarios never really worked very well so when all the old campaigns were tossed out with 3.0 none of the new ones recreated those complicated networks.